Source/Manager.swift

```diff
@@ -281,6 +284,11 @@ public class Manager {
             didReceiveChallenge challenge: NSURLAuthenticationChallenge,
             completionHandler: ((NSURLSessionAuthChallengeDisposition, NSURLCredential?) -> Void))
         {
+            guard sessionDidReceiveChallengeWithCompletion == nil else {
+                sessionDidReceiveChallengeWithCompletion?(session, challenge, completionHandler)
```

Comment:
shouldn't the `?` be a `!` here as `sessionDidReceiveChallengeWithCompletion` must be `!=nil`?

A way to let the compiler know that would be to use

```objc
if let didReceiveChallengeHandler == sessionDidReceiveChallengeWithCompletion {
    didReceiveChallengeHandler(session, challenge, completionHandler)
    return
}
```

Not sure whether this would be better. Has the advantage of not using `!` but the disadvantage of not ensuring we have `return` in the if-block.
So I'd probably stay with the guard but use `!` to get a runtime error instead of simply nothing happening (and the completion block never being called) 

Same argument in the case of `taskWillPerformHTTPRedirectionWithCompletion`, `taskDidReceiveChallengeWithCompletion`, `taskNeedNewBodyStreamWithCompletion `, `dataTaskDidReceiveResponseWithCompletion` and `dataTaskWillCacheResponseWithCompletion` of course